### DISCUSSION TITLE
Why can't the teleport application (web app) function work like a normal reverse proxy?

### DISCUSSION THREAD

**Discussion by caffeinism** [gravitational/teleport#19699] [Q&A] [answered]
The first time I saw teleport, I thought, "It would be great if it worked as a reverse proxy for a web app!" But well, it doesn't seem to work that way and requires a "broker" in the middle.

Why is this not designed to work with internal domains or private IPs?

**Comment by webvictim** [comment, marked as answer]
I'm not 100% sure I understand the premise of your question.

Teleport Application Access _is_ essentially an authenticated HTTP/HTTPS reverse proxy - what's the broker component that you're talking about? What's your use case? I highly suspect that if you're trying to give users authenticated access to a private HTTP endpoint, Teleport does do exactly what you want it to.

**Reply by caffeinism** [reply]
It seems I had some misunderstanding. From the guide to teleport, it seems that teleport runs once more on the host where the app is served. For example `Me - Internet - Teleport (reverse proxy) - Teleport (app proxy) - App`
I stupidly thought, "Why do I need a broker(app proxy) like that when I can connect directly to the app from Teleport (reverse proxy)?" 
However, I realized that I was misunderstood when I found out that the app can also be configured in the Teleport(reverse proxy and also app proxy) agent. I was forgetting the question when I found out about this fact.

(app proxy is also a reverse proxy, but please understand that the exposed proxy is a reverse proxy, and the "Web app access" function is written as an app proxy)

Thank you for the reply

**Reply by webvictim** [reply]
That makes sense - thanks for filling in the gaps.

I believe that our docs are currently being improved on to make it somewhat clearer that it's possible to run the Teleport proxy service and app service in the same process if desired, which should help prevent this sort of confusion in future!

**Reply by caffeinism** [reply]
Also, Teleport's web dashboard is structured to be difficult to use in this way. It guides only the method of the aforementioned tutorial. It would be much more convenient if you could configure it by directly entering the destination on the web page.
